logger/unilog_test.go

```diff
@@ -51,24 +53,38 @@ func TestReadlinesWithLongLines(t *testing.T) {
 	}
 }
 
+// double the first two instances of the character "e"
+type doubleEFilter struct{}
+
+func (ee *doubleEFilter) FilterLine(line string) string {
+	return strings.Replace(line, "e", "ee", 2)
+}
+
+func (ee *doubleEFilter) FilterJSON(line *json.LogLine) {
+	msg := (*line)["message"].(string)
+	(*line)["message"] = strings.Replace(msg, "e", "ee", 2)
+}
+
+type isToAintFilter struct{}
+
+func (ee *isToAintFilter) FilterLine(line string) string {
+	return strings.Replace(line, "is", "ain't", -1)
+}
+
+func (ee *isToAintFilter) FilterJSON(line *json.LogLine) {
+	msg := (*line)["message"].(string)
```

Comment:
We should be careful with filters that we don't restrict them to just one key's value; if we're logging structured data, we don't want to fail to apply filters to things that we'd otherwise expect them to apply to